coreneuron/io/reports/report_handler.cpp

```diff
@@ -13,18 +13,40 @@
 
 namespace coreneuron {
 
-void ReportHandler::create_report(double dt, double tstop, double delay) {
+std::vector<int> intersection_gids(const NrnThread& nt, std::vector<int>& target_gids) {
```

Comment:
maybe a stupid question, but you sure those should be `int`s and not `std::size_t` ?

Reply:
indeed at the end in libsonata-report we have a uint64_t for the node_id 
```
int sonata_add_node(const char* report_name,
                    const char* population_name,
                    uint64_t population_offset,
                    uint64_t node_id) {
```
However in neurodamus we write int from the CoreConfig.mod
```
int *gids = (int*) calloc(num_gids, sizeof(int));
...
fwrite(gids, sizeof(int), num_gids, fp);
```
And then the gids that each nrnthread has are also stored as an int:
```
class PreSyn: public ConditionEvent {
    ...
    int gid_{-1};
```
Maybe we can convert those to uint64_t for the intersection vector?

Reply:
Until we go to human brain scales I guess it's okay the gids to be `int`s?

Reply:
Ah, so in sonata we are already storing the gids as `uint_64` ? Then probably having all in that type would be more consistent - but only danger here is that we end up needing more memory and we're already fighting this...